jbi/actions/default.py

```diff
@@ -63,41 +63,37 @@ def groups2operation(steps):
     return by_operation
 
 
-def init(
-    jira_project_key,
-    steps: Optional[dict[str, list[str]]] = None,
-    **kwargs,
-):
-    """Function that takes required and optional params and returns a callable object"""
-    # Merge specified steps with default ones.
-    steps = {**DEFAULT_STEPS, **(steps or {})}
-
-    steps_by_operation = groups2operation(steps)
-
-    # Turn the steps strings into references to functions of the `jbi.actions.steps` module.
-    steps_callables = {
-        group: [getattr(steps_module, step_str) for step_str in steps_list]
-        for group, steps_list in steps_by_operation.items()
-    }
-
-    return Executor(jira_project_key=jira_project_key, steps=steps_callables, **kwargs)
+def init(parameters: ActionParams):
+    return Executor(parameters)
 
 
 class Executor:
     """Callable class that encapsulates the default action."""
 
-    def __init__(self, steps: dict[Operation, list[Callable]], **parameters):
+    def __init__(self, parameters: ActionParams):
         """Initialize Executor Object"""
-        self.steps = steps
         self.parameters = parameters
+        self.steps = self._initialize_steps(parameters.steps)
+
+    def _initialize_steps(self, steps: Optional[dict[str, list[str]]] = None):
+        steps = {**DEFAULT_STEPS, **(steps or {})}
+
+        steps_by_operation = groups2operation(steps)
+
+        # Turn the steps strings into references to functions of the `jbi.actions.steps` module.
+        steps_callables = {
+            group: [getattr(steps_module, step_str) for step_str in steps_list]
```

Comment:
Could this be done at the model level, so that we can catch unknown steps while parsing config? Ref #587

Reply:
I think this has to be done inside the class, no? Since the params of the action will determine what step functions are grabbed from the `steps` module.

I would like to actually move this to the `Action` model (as a validator maybe) so that we can't even initialize the models if something is misconfigured. But I'd like to do that in another PR focused on #587 if possible.

Reply:
You're right, this is #587